Makefile

```diff
@@ -0,0 +1,30 @@
+help:
+	@echo "Please use \`make <target>' where <target> is one of:"
+	@echo "  help            to show this message"
+	@echo "  all             to to execute lint and test-coverage"
+	@echo "  install         to install integrade to run integration tests"
+	@echo "  install-dev     to install integrade in editable mode to develop"
+	@echo "                  test cases"
+	@echo "  lint            to lint the source code"
+	@echo "  test            to run integrade's framework unit tests"
+	@echo "  test-coverage   to run integrade's unit tests and measure test"
+	@echo "                  coverage"
+
+all: lint test-coverage
+
+install:
+	pip install .
+
+install-dev:
+	pip install -e .[dev]
+
+lint:
+	flake8 .
+
+test:
+	py.test tests
+
+test-coverage:
+	py.test --verbose --cov-report term --cov=integrade tests
+
+.PHONY: all install install-dev lint test test-coverage
```

Comment:
Oooh, this is neat! I didn't know about `.PHONY` and had to look it up. If I understand correctly [what I read](https://stackoverflow.com/a/2145605), this allows these commands to work even if there's a file with the same name. I may have to steal this for one of our other make files.

Reply:
Yeah, that is correct. I am glad you found it useful.